codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -107,13 +104,14 @@ let checkContractIsInCurrentRegistrations = (
 ) => {
   switch dynamicContractRegistrations {
   | Some(dynamicContracts) =>
-    dynamicContracts.registrations->Array.some(d =>
-      d.dynamicContracts->Array.some(d =>
-        d.chainId == chain->ChainMap.Chain.toChainId &&
-        d.contractType == contractType &&
-        d.contractAddress == contractAddress
+    dynamicContracts.dynamicContractsByChain
+    ->Utils.Dict.dangerouslyGetNonOption(chain->ChainMap.Chain.toString)
+    ->Option.mapWithDefault(false, dcs =>
+      dcs->Array.some(dc =>
+        dc.contractType == contractType && dc.contractAddress == contractAddress
```

Comment:
I actually think we need to remove this check. I removed it in all the other places I think this was left by mistake. We are not allowing the same address registered to multiple contracts.
```suggestion
        dc.contractAddress == contractAddress
```